slam_toolbox/lib/karto_sdk/include/karto_sdk/Karto.h

```diff
@@ -2171,6 +2171,15 @@ namespace karto
       return m_Position.SquaredDistance(rOther.m_Position);
     }
 
+    /**
+     * Return the heading difference between two Pose2 (in radians)
+     * @return heading difference
+     */
+    inline kt_double HeadingDifference(const Pose2& rOther) const
```

Comment:
https://github.com/ros/angles/blob/ros2/angles/include/angles/angles.h#L84-L107

Implement as so, you even have a `math::NormalizeAngle` function